src/cluecode/finder_data.py

```diff
@@ -212,6 +227,11 @@ def classify(s, data_set, suffixes=None):
     if not s:
         return False
     s = s.lower().strip('/')
+    # Separate test for emails - need to ignore xyz@some.com, but not say, xyz@gruesome.com
+    if '@' in s and exact_match:
+        email_domain = s.split('@')
```

Comment:
Is it possible that there is more than on @ in that string? and if yes, would the segment on the right always be the domain?
What about using a `'.rpartition()` instead of `split()` and using proper names (e.g. host_name) for each part?

Reply:
Looking at the [Regular Expression for an email](https://en.wikipedia.org/wiki/Email_address#Syntax), the @ symbol is recommended to be used only once, however it can be used again within a quoted string. Short answer, yes, there could be a case where the last @ symbol is not followed by a domain, though highly unlikely. Do I have to assume we will be looking at emails such as those? 
> What about using a '.rpartition() instead of split() and using proper names (e.g. host_name) for each part?
Sure, that should do the job, provided the last @ is followed by a domain (a fairly strong assumption.)